docs/notes/2.23.x.md

```diff
@@ -147,6 +147,8 @@ The `python_aws_lambda_function` and `python_aws_lambda_layer` targets now allow
 
 The `python_google_cloud_function` target can now automatically select the `complete_platforms` used for building PEX files based on the value of the `runtime` field.
 
+Deprecate implicit resolution of PEX `platform` values for Function-as-a-Service targets, now that `complete_platforms` are available for both AWS Lambda and Google Cloud Functions.
```

Comment:
Users may have code that is currently working using this, so I think we should make the transition feel nicer if we can:

- explain why it is being deprecating (using the `--platform` argument to PEX is a very coarse summary of the target platform and regularly results in choosing incorrect native dependencies)
- link to the relevant docs 

I think we could also holistically consider the FaaS changelog items together, since it's all touching this concept. For instance:

```markdown
The function-as-a-service targets like `python_google_cloud_function`, `python_aws_lambda_function`, and `python_aws_lambda_layer` have had many changes to how they handle platforms:

- **Deprecated**: implicit platform inference. Previously, Pants would infer a `--platform` argument to PEX for target's without `complete_platforms` and with a runtime that doesn't have a built-in default complete platforms files. The `--platform` argument is a coarse summary of the target platform that regularly results in choosing incorrect native dependencies. In these cases, Pants now prompts one to [provide a value for the `complete_platforms` field manually](...).
- To make this easier, `python_google_cloud_function` target will now automatically select a `complete_platforms` based on the value of the `runtime` field, similar to the AWS targets.
- ... architecture ...
- ... `file` and `resource` ...
```

(I'm putting the deprecation first since that's what's affects existing users most directly.)

What do you think?